catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3ExceptionMapper.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.files.s3;
+
+import java.util.OptionalInt;
+import org.projectnessie.catalog.files.api.ObjectIOExceptionMapper;
+import software.amazon.awssdk.core.exception.SdkServiceException;
+
+public class S3ExceptionMapper implements ObjectIOExceptionMapper.Analyzer {
+
+  public static final S3ExceptionMapper INSTANCE = new S3ExceptionMapper();
+
+  private S3ExceptionMapper() {}
+
+  @Override
+  public OptionalInt httpStatusCode(Throwable th) {
+    if (th instanceof SdkServiceException) {
```

Comment:
Wanna handle `SdkClientException` as well?

Reply:
`SdkClientException` can happen only before the storage service is contacted... Can we really do better than a "Server Failure" in this case?

Reply:
Sure, it can happen before. But IIUC that can also happen for misconfigured clients, and user configuration mistakes.

Reply:
In the latest code those exceptions will get the `UNKNOWN` status code and be represented as Internal Server Errors (500) at the REST layer.

Reply:
We need HTTP 600 error codes - server failed because of user configuration mistake :troll: